edgedb/datatypes/object.c

```diff
@@ -189,6 +189,10 @@ object_hash(EdgeObject *o)
 static PyObject *
 object_getattr(EdgeObject *o, PyObject *name)
 {
+    // Used in `dataclasses.as_dict()`
+    if (PyUnicode_CompareWithASCIIString(name, "__dataclass_fields__") == 0) {
+        return EdgeRecordDesc_GetDataclassFields((PyObject *)o->desc);
+    }
```

Comment:
I think this should be moved into the `L_NOT_FOUND` case. Pointless to call this on every other attribute lookup.